Sources/MapboxDirections/Directions.swift

```diff
@@ -375,6 +384,103 @@ open class Directions: NSObject {
+                }
+                return
+            }
+            
+            DispatchQueue.global(qos: .userInitiated).async {
+                do {
+                    let decoder = JSONDecoder()
+                    decoder.userInfo = [.refreshingRoute: routes[routeIndex],
+                                        .routeIndex: routeIndex,
+                                        .legIndex: currentLegIndex,
+                                        .credentials: self.credentials]
+                    
+                    guard let disposition = try? decoder.decode(ResponseDisposition.self, from: data) else {
+                        let apiError = DirectionsError(code: nil, message: nil, response: possibleResponse, underlyingError: possibleError)
+
+                        DispatchQueue.main.async {
+                            completionHandler(self.credentials, .failure(apiError))
+                        }
+                        return
+                    }
+                    
+                    guard (disposition.code == nil && disposition.message == nil) || disposition.code == "Ok" else {
+                        let apiError = DirectionsError(code: disposition.code, message: disposition.message, response: response, underlyingError: possibleError)
+                        DispatchQueue.main.async {
+                            completionHandler(self.credentials, .failure(apiError))
+                        }
+                        return
+                    }
+                    
+                    let result = try decoder.decode(RouteRefreshResponse.self, from: data)
+                    guard result.route != nil else {
+                        DispatchQueue.main.async {
+                            completionHandler(self.credentials, .failure(.unableToRefresh))
+                        }
+                        return
+                    }
+                    
+                    DispatchQueue.main.async {
+                        completionHandler(self.credentials, .success(result))
+                    }
+                } catch {
+                    DispatchQueue.main.async {
+                        let bailError = DirectionsError(code: nil, message: nil, response: response, underlyingError: error)
+                        completionHandler(self.credentials, .failure(bailError))
+                    }
+                }
+            }
+        }
+        requestTask.priority = 1
+        requestTask.resume()
+        return requestTask
+    }
+    
+    open func urlRequest(forRefreshing routeId: String, routeIndex: Int, currentLegIndex: Int) -> URLRequest {
+        var params: [URLQueryItem] = []
+        params += [URLQueryItem(name: "access_token", value: credentials.accessToken)]
+        
+        var unparameterizedURL = URL(string: "directions-refresh/v1/\(DirectionsProfileIdentifier.automobileAvoidingTraffic.rawValue)", relativeTo: credentials.host)!
```

Comment:
The `automobileAvoidingTraffic` profile identifier is hard-coded here because the directions refresh API endpoint only supports refreshing routes calculated using this profile. (The other profiles have no traffic congestion data to refresh.) There is code in RouteOptions to avoid enabling route refreshing for the other profiles, but should we also check here that the `RouteLeg.profileIdentifier` is `.automobileAvoidingTraffic`, in case the developer enabled route refreshing on a route of the route profile?